xarray/core/indexing.py

```diff
@@ -1288,6 +1290,47 @@ def __init__(self, array):
         self.array = array
 
 
+class ArrayApiIndexingAdapter(ExplicitlyIndexedNDArrayMixin):
+    """Wrap an array API array to use explicit indexing."""
+
+    __slots__ = ("array",)
+
+    def __init__(self, array):
+        if not hasattr(array, "__array_namespace__"):
+            raise TypeError(
+                "ArrayApiIndexingAdapter must wrap an object that "
+                "implements the __array_namespace__ protocol"
+            )
+        self.array = array
+
+    def __getitem__(self, key):
+        if isinstance(key, BasicIndexer):
+            return self.array[key.tuple]
+        elif isinstance(key, OuterIndexer):
+            # manual orthogonal indexing (implemented like DaskIndexingAdapter)
+            key = key.tuple
+            value = self.array
+            for axis, subkey in reversed(list(enumerate(key))):
+                value = value[(slice(None),) * axis + (subkey, Ellipsis)]
+            return value
+        else:
+            assert isinstance(key, VectorizedIndexer)
+            raise TypeError("Vectorized indexing is not supported")
+
+    def __setitem__(self, key, value):
+        if isinstance(key, BasicIndexer):
+            self.array[key.tuple] = value
+        elif isinstance(key, OuterIndexer):
+            self.array[key.tuple] = value
```

Comment:
unless i'm missing something, we should be able to combine these two branches into one, correct? 

```suggestion
        if isinstance(key, (BasicIndexer, OuterIndexer)):
            self.array[key.tuple] = value
```

Reply:
Correct - fixed.